runtime/ms_rest_azure/lib/ms_rest_azure/resource.rb

```diff
@@ -23,6 +23,15 @@ class Resource
     # @return [Hash{String => String}] the tags attached to resources (optional).
     attr_accessor :tags
 
+    # @return [String] the name of the resource group of the resource.
+    def resource_group

```

Comment:
One more optimization thought is as `resource_group` is a computed property of `Resource` class, we should only compute it once when we are setting up `id` not every time we access the `resource_group`.